src/runtime/runtime-basic.js

```diff
@@ -0,0 +1,200 @@
+
+  // Create the public version of the SwG Basic runtime.
+  const publicBasicRuntime = createPublicBasicRuntime(basicRuntime);
+
+  /**
+   * Executes a callback when the runtime is ready.
+   * @param {function(!BasicSubscriptions)} callback
+   */
+  function callWhenRuntimeIsReady(callback) {
+    if (!callback) {
+      return;
+    }
+
+    basicRuntime.whenReady().then(() => {
+      callback(publicBasicRuntime);
+    });
+  }
+
+  // Queue up any callbacks the publication might have provided.
+  const waitingCallbacks = [].concat(win[RUNTIME_BASIC_PROP]);
+  waitingCallbacks.forEach(callWhenRuntimeIsReady);
+
+  // If any more callbacks are `push`ed to the global SwG Basic variables,
+  // they'll be queued up to receive the SwG Basic runtime when it's ready.
+  win[RUNTIME_BASIC_PROP] = {
+    push: callWhenRuntimeIsReady,
+  };
+
+  // Set variable for testing.
+  basicRuntimeInstance_ = basicRuntime;
+
+  // Automatically set up buttons and auto prompt.
+  basicRuntime.setupButtons();
+  basicRuntime.setupAutoPrompt();
+}
+
+/**
+ * @implements {BasicSubscriptions}
+ */
+class BasicRuntime {
+  /**
+   * @param {!Window} win
+   */
+  constructor(win) {
+    /** @private @const {!Window} */
+    this.win_ = win;
+
+    /** @private @const {!Promise} */
+    this.ready_ = Promise.resolve();
+  }
+
+  /**
+   * @return {!Promise}
+   */
+  whenReady() {
+    return this.ready_;
+  }
+
+  /**
+   * @param {boolean} commit
+   * @return {!Promise<!ConfiguredBasicRuntime>}
+   * @private
+   */
+  configured_(commit) {
+    if (!this.committed_ && commit) {
+      // TODO(stellachui): Read the PageConfig from the markup and create the
+      //   ConfiguredBasicRuntime object once that's done.
+    }
+    return Promise.resolve();
+  }
+
+  /** @override */
+  /* eslint-disable no-unused-vars */
+  init({
+    type,
+    isAccessibleForFree,
+    isPartOfType,
+    isPartOfProductId,
+    autoPromptType = AutoPromptType.NONE,
+  } = {}) {
+    // TODO(stellachui): Generate the markup.
+    this.configured_(true);
+  }
+  /* eslint-enable no-unused-vars */
+
+  /**
+   * Sets up all the buttons on the page with attribute
+   * 'swg-standard-button:subscriptions' or 'swg-standard-button:contributions'.
+   */
+  setupButtons() {
+    return this.configured_(true).then((runtime) => runtime.setupButtons());
+  }
+
+  /**
+   * Sets up an auto prompt, as configured on the page's structured markup.
+   */
+  setupAutoPrompt() {
+    return this.configured_(true).then((runtime) => runtime.setupAutoPrompt());
+  }
+}
+
+/**
+ * @implements {BasicSubscriptions}
+ */
+// eslint-disable-next-line no-unused-vars
+class ConfiguredBasicRuntime {
+  /**
+   * @param {!Window|!Document|!Doc} winOrDoc
+   * @param {!../model/page-config.PageConfig} pageConfig
+   */
+  constructor(winOrDoc, pageConfig) {
+    this.configuredClassicRuntime_ = new ConfiguredRuntime(
+      winOrDoc,
+      pageConfig,
+      /* integr= */ null,
+      /* config= */ null
```

Comment:
null is not needed here, but aren't integr and config needed?

Reply:
They're used for the config the config promise, and if unspecified, still be set to default. I am following up with Shams re:whether we need to make certain things configurable, so this value may change in the future.